### DISCUSSION TITLE
Issue on Hook

### DISCUSSION THREAD

**Discussion by ThemeAZ** [impress-org/givewp#7358] [Developer Q&A]
Hi Support team,

Why I use this hook for check donate submit form: when use **_one time donate form_** is working fine, but on **_recurring donate form_** is not work?

`add_action('givewp_donation_updated', 'custom_after_donation', 10, 1);`

Any help en this issue?

Thanks,
Lai

**Comment by 007amrit** [comment]
Hey 

I had an exact question but I didn't find anything 

So I end up using this 

it will run each time for any kind of transaction whether pending , cancelled, recurring  published

so you need to find it out using new status 



add_action( 'give_update_payment_status', 'donation_action_function', 200, 3 );

function donation_action_function( $donation_id, $new_status, $old_status ) {
    
    $accepted_statuses = ['publish', 'give_subscription'];
  
    if(in_array($new_status, $accepted_statuses)){
       // do your action here
    }

}


in case you find better solution please share 

thanks
